src/psyclone/tests/psyir/transformations/inline_trans_test.py

```diff
@@ -772,6 +772,172 @@ def test_apply_multi_function(fortran_reader, fortran_writer, tmpdir):
+    ("", "", ""),
+    ("module test_mod\ncontains\n", "end module test_mod\n", "  "),
+    ("module test_mod\nuse dummy\ncontains\n", "end module test_mod\n", "  ")])
+def test_apply_raw_subroutine(
+        fortran_reader, fortran_writer, tmpdir, start, end, indent):
+    '''Test the apply method works correctly when the routine to be
+    inlined is a raw subroutine and is called directly from a call
+    from another raw subroutine, a call from a subroutine within a
+    module but without a use statement and a call from a subroutine
+    within a module with a wildcard use statement.
+
+    '''
+    code = (
+        f"{start}"
+        f"  subroutine run_it()\n"
+        f"    real :: a\n"
+        f"    call sub(a)\n"
+        f"  end subroutine run_it\n"
+        f"{end}"
+        f"subroutine sub(x)\n"
+        f"  real, intent(inout) :: x\n"
+        f"  x = 2.0*x\n"
+        f"end subroutine sub\n")
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    inline_trans.apply(routine)
+    output = fortran_writer(psyir)
+    expected = (
+        f"{indent}subroutine run_it()\n"
+        f"{indent}  real :: a\n\n"
+        f"{indent}  a = 2.0 * a\n\n"
+        f"{indent}end subroutine run_it\n")
+    assert expected in output
+    if "use dummy" not in output:
+        # Compilation will not work with "use dummy" as there is no
+        # mod file.
+        assert Compile(tmpdir).string_compiles(output)
+
+
+@pytest.mark.parametrize("use1, use2", [
+    ("use inline_mod, only : sub\n", ""), ("use inline_mod\n", ""),
+    ("", "use inline_mod, only : sub\n"), ("", "use inline_mod\n")])
+def test_apply_container_subroutine(
+        fortran_reader, fortran_writer, tmpdir, use1, use2):
+    '''Test the apply method works correctly when the routine to be
+    inlined is in a different container and is within a module (so
+    a use statement is required).
+
+    '''
+    code = (
+        f"module inline_mod\n"
+        f"contains\n"
+        f"  subroutine sub(x)\n"
+        f"    real, intent(inout) :: x\n"
+        f"    x = 2.0*x\n"
+        f"  end subroutine sub\n"
+        f"end module inline_mod\n"
+        f"module test_mod\n"
+        f"{use1}"
+        f"contains\n"
+        f"  subroutine run_it()\n"
+        f"    {use2}"
+        f"    real :: a\n"
+        f"    call sub(a)\n"
+        f"  end subroutine run_it\n"
+        f"end module test_mod\n")
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    inline_trans.apply(routine)
+    output = fortran_writer(psyir)
+    assert (
+        "    real :: a\n\n"
+        "    a = 2.0 * a\n\n"
+        "  end subroutine run_it" in output)
+    assert Compile(tmpdir).string_compiles(output)
+
+
+def test_apply_container_subroutine_error(fortran_reader):
+    '''Test the apply method does not find the Routine if the Call is
```

Comment:
I think it would be better if you followed your usual practice of testing the various utility search routines separately (plus one check that they are called by the transformation).

Reply:
That took a while, but now done.

Reply:
Thanks, I think it was worth it. (Easy for me to say that though.)